### PR TITLE
Avoid running cargo+internal lints when not enabled

### PR DESCRIPTION
**Pull request by flip1995**
r? @matthiaskrgr 

changelog: none

### CODE REVIEW

File: clippy_lints/src/cargo_common_metadata.rs [hunk 1]
```diff
@@ -59,8 +59,12 @@ fn is_empty_vec(value: &[String]) -> bool {
 
 declare_lint_pass!(CargoCommonMetadata => [CARGO_COMMON_METADATA]);
 
-impl EarlyLintPass for CargoCommonMetadata {
-    fn check_crate(&mut self, cx: &EarlyContext<'_>, _: &Crate) {
+impl LateLintPass<'_, '_> for CargoCommonMetadata {
```
Comment (flip1995) [inline clippy_lints/src/cargo_common_metadata.rs:62]: Downside: The cargo lints are now `LateLintPass`es

**Comment by lzutao** [comment]
What's the motivation of this change?

**Comment by matthiaskrgr** [comment]
@lzutao  when running `clippy-driver foo.rs` in a directory that had a big `Cargo.toml` somewhere in a parent path, clippy-driver would redundantly search for the nearest cargo manifest run `clippy::cargo`-lints which invoced `cargo-metadata` on the entire crate, slowing everything down significantly depending on the project complexity.
https://discordapp.com/channels/442252698964721669/459149871191031810/701877035936645192

https://discordapp.com/channels/442252698964721669/459149871191031810/702159543538417714

When I tested a version of this branch, it managed to speed up my testcase (running clippy-driver on testcases inside the rustc repo) by ~10x :)

**Comment by matthiaskrgr** [comment]
Thanks a lot @flip1995  !
Feel free to r=me once github is working again :)

**Comment by flip1995** [comment]
To add to this: I ran flamegraph on `clippy-driver`, when run inside the clippy repo dir on a single (hello-world) file. Without this change the cargo metadata collection took ~30-40% of the runtime.

@bors r=matthiaskrgr

**Comment by matthiaskrgr** [comment]
````
   Doc-tests clippy_lints
error: unnecessary braces around block return value
   |
note: the lint level is defined here
  --> /home/runner/work/rust-clippy/rust-clippy/clippy_lints/src/lib.rs:12:45
   |
12 | #![cfg_attr(feature = "deny-warnings", deny(warnings))]
   |                                             ^^^^^^^^
   = note: `#[deny(unused_braces)]` implied by `#[deny(warnings)]`
error: unnecessary braces around block return value
error: aborting due to 2 previous errors
error: test failed, to rerun pass '--doc'
````

could this be side-effect of changing pass order?

**Pull request merged**